### PR TITLE
Added Support for Setting Device Token before SDK Initialization as well

### PR DESCRIPTION
**Pull request by desusai7**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-ios/101)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,15 +39,15 @@ The iOS SDK is available through [**CocoaPods**](https://cocoapods.org), [**Cart
 To install the SDK, simply add the following line to your Podfile:
 
 ```xcode
-pod 'Rudder', '1.1.5'
+pod 'Rudder', '1.2.5'
```
Comment (pallabmaiti) [inline README.md:42]: Version should be 1.1.6

File: Rudder.podspec [hunk 2]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = 'Rudder'
-  s.version          = "1.1.5"
+  s.version          = "1.2.5"
```
Comment (pallabmaiti) [inline Rudder.podspec:3]: Here.

**Review by pallabmaiti — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (desusai7) [reply README.md]: Changed version to 1.2.0 as we discussed

File: Sources/Classes/RSClient.m [hunk 3]
```diff
@@ -374,4 +379,16 @@ + (void)setAnonymousId:(NSString *)anonymousId {
     [preferenceManager saveAnonymousId:anonymousId];
 }
 
++ (void)putDeviceToken:(NSString *)deviceToken {
+    if(deviceToken != nil && [deviceToken length] != 0)
+    {
```
Comment (pallabmaiti) [inline Sources/Classes/RSClient.m:384]: Can you fix the the bracket alignment? Should be start in previous line.

File: Sources/Classes/Public/RSClient.h [hunk 4]
```diff
@@ -76,6 +76,7 @@ NS_ASSUME_NONNULL_BEGIN
 - (RSContext *) getContext;
 
 + (void) setAnonymousId: (NSString *__nullable) anonymousId;
++ (void) putDeviceToken: (NSString *) deviceToken;
```
[the remark this answers is not in the document]
Comment (desusai7) [reply Sources/Classes/Public/RSClient.h:79]: Yeah, named it that way in correspondence with Android SDK

**Review by pallabmaiti — DISMISSED** [review]

**Review by pallabmaiti — APPROVED** [review]

**Pull request merged**